nanovdb/nanovdb/NanoVDB.h

```diff
@@ -6509,6 +4822,89 @@ using OnIndexGrid = Grid<OnIndexTree>;
 using IndexMaskGrid = Grid<IndexMaskTree>;
 using OnIndexMaskGrid = Grid<OnIndexMaskTree>;
 
+// --------------------------> callNanoGrid <------------------------------------
+
+/**
+* @brief Below is an example of the struct used for generic programming with callNanoGrid
+* @details For an example see "struct Crc32TailOld" in nanovdb/tools/GridChecksum.h or
+*          "struct IsNanoGridValid" in nanovdb/tools/GridValidator.h
+* @code
+*   struct OpT {
+        // define these two static functions with non-const GridData
+*       template <typename BuildT>
+*       static auto   known(      GridData *gridData, args...);
+*       static auto unknown(      GridData *gridData, args...);
+*       // or alternatively these two static functions with const GridData
+*       template <typename BuildT>
+*       static auto   known(const GridData *gridData, args...);
+*       static auto unknown(const GridData *gridData, args...);
+*   };
+*  @endcode
+*
+* @brief Here is an example of how to use callNanoGrid in client code
+* @code
+*    return callNanoGrid<OpT>(gridData, args...);
+* @endcode
+*/
+
+/// @brief Use this function, which depends a pointer to GridData, to call
+///        other functions that depend on a NanoGrid of a known ValueType.
+/// @details This function allows for generic programming by converting GridData
+///          to a NanoGrid of the type encoded in GridData::mGridType.
+template<typename OpT, typename GridDataT, typename... ArgsT>
+auto callNanoGrid(GridDataT *gridData, ArgsT&&... args)
```

Comment:
Would it make sense to change the name of this function from `callNanoGrid` to `processTypedGrid`? The argument for changing the name to `processTypedGrid` is that it provides a similar functionality as the one in openvdb, cf. this [documentation]( https://www.openvdb.org/documentation/doxygen/codeExamples.html#sGenericProg). The argument against changing the name is because it expects a different 'signature' on the operator that a user needs to pass.